ui/desktop/tests/acceptance/projects/sessions/session-test.js

```diff
@@ -148,13 +148,126 @@ module('Acceptance | projects | sessions | session', function (hooks) {
+                deletion_time: '',
+                destroyed: false,
+                version: 8,
+              },
+            },
+          },
+        },
+      ],
+    });
+
+    await visit(urls.target);
+    await click(TARGET_CONNECT_BUTTON);
+    assert.strictEqual(currentURL(), urls.session);
+
+    assert.dom('.credential-panel-header').exists();
+    assert.dom('.credential-panel-body').exists();
+    assert.dom('.credential-secret').exists();
+
+    // Check if nested data is displayed in a key/value format without escape characters
+    assert
+      .dom('.secret-container:nth-of-type(1)')
+      .hasText('backslash ■ ■ ■ ■ ■ ■ ■ ■ ■ ■ ■ ■');
+    await click('.secret-container:nth-of-type(1) .hds-icon');
+    const expectedOutput = String.raw`password\withslash\ `;
+    assert
+      .dom('.secret-container:nth-of-type(1) .secret-content')
+      .hasText(expectedOutput);
+
+    assert
+      .dom('.secret-container:nth-of-type(2)')
+      .hasText('email.address ■ ■ ■ ■ ■ ■ ■ ■ ■ ■ ■ ■');
+    await click('.secret-container:nth-of-type(2) .hds-icon');
+    assert
+      .dom('.secret-container:nth-of-type(2) .secret-content')
+      .hasText('test.com');
+  });
+
+  test('visiting session with static type credentials should display nested data in a key/value format without escape characters', async function (assert) {
+    this.ipcStub.withArgs('cliExists').returns(true);
+    this.ipcStub.withArgs('connect').returns({
+      session_id: instances.session.id,
+      address: 'a_123',
+      port: 'p_123',
+      protocol: 'tcp',
+      credentials: [
+        {
+          credential_source: {
+            id: 'credjson_7cKBbBEkC3',
+            credential_store_id: 'csst_yzlsot2pum',
+            type: 'static',
+            credential_type: 'json',
+          },
+          secret: {
+            raw: 'eyJhcnJheSI6WyJvbmUiLCJ0d28iLCJ0aHJlZSIsIm9uZSIsInR3byIsInRocmVlIiwib25lIiwidHdvIiwidGhyZWUiLCJvbmUiLCJ0d28iLCJ0aHJlZSJdLCJuZXN0ZWQiOnsiYm9vbCI6dHJ1ZSwibG9uZyI6IjEyMjM1MzQ1NmFzZWRmYTQzd3J0ZjIzNGYyM2FzZGdmYXNkZnJnYXdzZWZhd3NlZnNkZjQiLCJzZWNlcmV0Ijoic28gbmVzdGVkIn0sInRlc3QiOiJwaHJhc2UifQ==',
+            decoded: {
+              nested_secret: {
+                complex_nest: {
+                  blackslash: 'password\\withslash\\',
+                },
+              },
+            },
+          },
+        },
+      ],
+    });
+
+    await visit(urls.target);
+    await click(TARGET_CONNECT_BUTTON);
+    assert.strictEqual(currentURL(), urls.session);
+
+    assert.dom('.credential-panel-header').exists();
+    assert.dom('.credential-panel-body').exists();
+    assert.dom('.credential-secret').exists();
+
+    // Check if nested data is displayed in a key/value format without escape characters
+    const expectedOutput = String.raw`password\withslash\ `;
+    assert
+      .dom('.secret-container:nth-of-type(1)')
+      .hasText('nested_secret.complex_nest.blackslash ■ ■ ■ ■ ■ ■ ■ ■ ■ ■ ■ ■');
```

Comment:
It seems weird to include the ■ characters, why not just use an `includesText` assert instead?

Reply:
good point, I changed to includes